src/core/swap.c

```diff
@@ -395,51 +394,42 @@ static int swap_setup_unit(
 
                 if (s->from_proc_swaps &&
                     !path_equal(s->parameters_proc_swaps.what, what_proc_swaps))
-                        return log_error_errno(SYNTHETIC_ERRNO(EEXIST),
-                                               "Swap %s appeared twice with different device paths %s and %s, refusing.",
-                                               e, s->parameters_proc_swaps.what, what_proc_swaps);
-
-                new = false;
+                        return log_unit_error_errno(u, SYNTHETIC_ERRNO(EEXIST),
+                                                    "Swap appeared twice with different device paths %s and %s, refusing.",
+                                                    s->parameters_proc_swaps.what, what_proc_swaps);
         } else {
-                new = true;
-
-                r = unit_new_for_name(m, sizeof(Swap), e, &u);
-                if (r < 0) {
-                        log_unit_warning_errno(u, r, "Failed to load swap unit: %m");
-                        goto fail;
-                }
+                r = unit_new_for_name(m, sizeof(Swap), e, &new);
+                if (r < 0)
+                        return log_warning_errno(r, "Failed to load swap unit '%s': %m", e);
 
+                u = new;
```

Comment:
Previously this was not cleaned up, now it is? Is that intentional?

Reply:
That was also a bug introduced by e9fa1bf704ad2f0a7e257e29889315118b0df459 :/